src/aap_eda/services/project/scm.py

```diff
@@ -322,3 +327,32 @@ def __call__(
                     raise ScmAuthenticationError(err_msg)
                 raise ScmError(f"{self.ERROR_PREFIX} {err_msg}")
             raise ScmError(f"{self.ERROR_PREFIX} {outputs.getvalue().strip()}")
+
+
+URL_REGEX = "^((git|ssh|http(s)?)|git\\+ssh|(git@[\\w\\.-]+))(:(//)?)([\\w\\.@\\:/\\-~]+)(\\.git)?(/)?$"  # noqa: E501
+
+
+def is_git_url_valid(url: str) -> bool:
+    if not re.match(URL_REGEX, url):
+        logger.error(f"Invalid SCM URL {url}")
+        return False
+    return True
+
+
+def is_refspec_valid(refspec: str, is_branch: bool) -> bool:
+    if is_branch:
+        args = [GIT_COMMAND, "check-ref-format", "--branch", refspec]
+    else:
+        args = [GIT_COMMAND, "check-ref-format", refspec]
+    with tempfile.TemporaryDirectory(prefix="EDA_REFSPEC") as home_dir:
+        result = subprocess.run(args, cwd=home_dir)
+        if result.returncode != 0:
+            logger.error(
+                f"git check-ref-format {refspec} failed with exit code "
+                f"{result.returncode}"
+            )
+            return False
+        if any(arg in refspec for arg in ["{{", "}}", "lookup"]):
```

Comment:
@bzwei Is this for branch test or ref spec. I think the git check-ref-format already says these are invalid with and without spaces.